### PR TITLE
Increase testing coverage

### PR DESCRIPTION
**Pull request by jmporchet**
# Summary

Working on #92. This is a WIP.
I am not really familiar with hooks and their testing, so I welcome suggestions on better test coverage!

## Test Plan
Initial test files, that will need to cover more features, especially VersionSelector, DiffViewer and UsefulContentSection.

## Checklist

- [ ] I tested this thoroughly
- [ ] I added the documentation in `README.md` (if needed)

### CODE REVIEW

File: src/__tests__/components/Home.spec.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { render } from '@testing-library/react'
+import Home from '../../components/pages/Home'
+
+it('renders without crashing', () => {
+  const { container } = render(<Home />)
+
+  expect(container).toMatchSnapshot()
+})
+
+it('renders without crashing', () => {
```
Comment (KevinHu2014) [inline src/__tests__/components/Home.spec.js]: I'm really new to `react-testing-library`.
Just wonder to know the different between `line 5-9` and `line 11-15` ?

Comment (lucasbento) [reply src/__tests__/components/Home.spec.js]: There isn't really a difference, the cases are just duplicated 😅

File: .gitignore [hunk 2]
```diff
@@ -27,4 +27,7 @@ yarn-error.log*
 
 # IDE
 .idea
-.vscode
\ No newline at end of file
+.vscode
+
+# testing
+/.jest
```
Comment (lucasbento) [inline .gitignore]: Never did this, what's this directory for?

File: src/__tests__/components/common/VersionSelector.spec.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { render, fireEvent } from '@testing-library/react'
+import VersionSelector from '../../../components/common/VersionSelector'
+
+it('renders without crashing', () => {
+  const { container } = render(<VersionSelector />)
+
+  expect(container).toMatchSnapshot()
+})
+
+// it('renders without crashing2', () => {
```
Comment (lucasbento) [inline src/__tests__/components/common/VersionSelector.spec.js]: This one needs to be removed before it gets merged, you can also put a good name to the test case and use `xit` to not run the test but keep the code you already wrote.

File: .gitignore (hunk 2 above)
Comment (jmporchet) [reply .gitignore]: jest will generate cache files, we don't really want it to pollute the repository

File: src/__tests__/components/common/VersionSelector.spec.js (hunk 3 above)
Comment (jmporchet) [reply src/__tests__/components/common/VersionSelector.spec.js]: definitely needs to be removed.
Can you give me more info about `xit`

File: src/__tests__/components/Home.spec.js (hunk 1 above)
Comment (jmporchet) [reply src/__tests__/components/Home.spec.js]: yep 😅

File: src/__tests__/components/common/VersionSelector.spec.js (hunk 3 above)
Comment (jmporchet) [reply src/__tests__/components/common/VersionSelector.spec.js]: cool, I looked it up and it's an alias for `it.skip`. This is awesome, I didn't know that existed!

**Comment by jmporchet** [comment]
It would be great to have at least one fully featured test with interaction on components.
For this, I have laid out a plan in `src/__tests__/components/common/VersionSelector.spec.js` for future contributors (me or someone else!) to add as I'm currently not familiar with hooks testing.

File: src/__tests__/components/common/VersionSelector.spec.js (hunk 3 above)
Comment (lucasbento) [reply src/__tests__/components/common/VersionSelector.spec.js]: I didn't know `it.skip` only `xit` ahah, good to know!

**Pull request merged**